### PR TITLE
SHOW-52: Add development docker-compose

### PR DESCRIPTION
**Pull request by markspolakovs**

### CODE REVIEW

File: cmd/main/main.go [hunk 1]
```diff
@@ -78,17 +79,31 @@ func main() {
 			Password: os.Getenv("ST_DB_PASSWORD"),
 		},
 	}
+	log.Printf("debug: configuration: %+v", conf)
```
Comment (markspolakovs) [inline cmd/main/main.go:82]: @markspolakovs remove this

File: cmd/main/main.go [hunk 2]
```diff
@@ -78,17 +79,31 @@ func main() {
...
+
+	upToDate, err := migrations.IsUpToDate(db.DB)
 	if err != nil {
-		log.Fatalf("unable to get schema version: %+v", err)
+		log.Fatalf("failed to check if migrations are up to date: %+v", err)
 	}
-	if schemaVersion != schema.HighestSchemaVersion {
-		log.Fatalf("Database schema outdated, please run 'init' again; expected %d, but found %d", schema.HighestSchemaVersion, schemaVersion)
+	if !upToDate {
+		log.Fatalf("database not up to date, please run 'init' or set ST_DB_AUTO_INIT=true")
```
Comment (rmil) [inline cmd/main/main.go:106]: Is it worth moving this to `db.New()`?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    image: postgres:alpine
+    volumes:
+      - pgdata:/var/lib/postgresql/data
+    environment:
+        POSTGRES_PASSWORD: showtime
+        POSTGRES_USER: showtime
+        POSTGRES_DB: showtime
+
+  brave:
+    image: registry.comp.ystv.co.uk/brave:latest
```
Comment (rmil) [inline docker-compose.yml:14]: To help people who don't have access to the registry, we could put a reference to the repo where they can build the image themselves? For this it's [github.com/ystv/brave](github.com/ystv/brave)

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    ports:
+      - "5000:5000"
+
+  nginx:
+    image: registry.comp.ystv.co.uk/nginx/nginx-rtmp:latest
```
Comment (rmil) [inline docker-compose.yml:19]: Outside the scope of this ticket, but if we do refactor how we make this image is built into a public repo, would be good to drop a link as well

File: docker/nginx.conf [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		access_log  /var/log/nginx/access.log;
+		listen 1935;
+
+		wait_key on;
+		wait_video on;
+		ping 60s;
+
+		#matthew 2019
+		drop_idle_publisher 30s;
+		idle_streams off;
```
Comment (rmil) [inline docker/nginx.conf:67]: These can be bunched up and the comment removed

**Review by rmil — COMMENTED** [review]
Looks good! Definitely needed !(I've just bodged and used stuff running on the ystv servers, fine for on-campus but a pickle for off-site)

Will this also make it easier to get a codespace going?

File: cmd/main/main.go (hunk 2 above)
Comment (markspolakovs) [reply cmd/main/main.go:106]: Problem is that `cmd/init` also uses `db.New()`, but there we want to skip the outdated check - do you think it's worth adding a flag to `db.New()` for that?

Comment (rmil) [reply cmd/main/main.go:106]: Could make it an option in `db.Config`?

**Comment by markspolakovs** [comment]
> Will this also make it easier to get a codespace going?

In theory

**Pull request merged**